pool/app/custom_field/repo/repo_partial_update.ml

```diff
@@ -0,0 +1,227 @@
+    |> Utils.Database.exec (Database.Label.value pool) request
+  in
+  match t with
+  | Boolean (_, answer) ->
+    answer
+    |> value_to_store is_admin
+    |> map_or ~clear (fun (id, value) ->
+         update_answer id (Utils.Bool.to_string value))
+  | MultiSelect (_, _, answer) ->
+    answer
+    |> value_to_store is_admin
+    |> map_or ~clear (fun (id, value) ->
+         value
+         |> CCList.map (fun { Entity.SelectOption.Public.id; _ } -> id)
+         |> Repo_entity.yojson_of_multi_select_answer
+         |> Yojson.Safe.to_string
+         |> update_answer id)
+  | Number (_, answer) ->
+    answer
+    |> value_to_store is_admin
+    |> map_or ~clear (fun (id, value) ->
+         update_answer id (CCInt.to_string value))
+  | Select (_, _, answer) ->
+    answer
+    |> value_to_store is_admin
+    |> map_or ~clear (fun (id, value) -> update_answer id (option_id value))
+  | Text (_, answer) ->
+    answer
+    |> value_to_store is_admin
+    |> map_or ~clear (fun (id, value) -> update_answer id value)
+;;
+
+let update pool user (field : PartialUpdate.t) (contact : Contact.t) =
+  let open Entity in
+  let base_caqti = Pool_common.Repo.Id.t in
+  let dyn =
+    Dynparam.empty
+    |> Dynparam.add base_caqti (contact |> Contact.id)
+    |> Dynparam.add Caqti_type.ptime (Ptime_clock.now ())
+  in
+  let update_user_table (dyn, sql) =
+    let open Caqti_request.Infix in
+    let (Dynparam.Pack (pt, pv)) = dyn in
+    let update_request = sql |> update_sql |> pt ->. Caqti_type.unit in
+    Utils.Database.exec (pool |> Pool_database.Label.value) update_request pv
+  in
+  let open PartialUpdate in
+  match field with
+  | Firstname (version, value) ->
+    let%lwt (_ : Sihl_user.t) =
+      update_sihl_user
+        pool
+        ~firstname:(value |> Pool_user.Firstname.value)
+        contact
+    in
+    ( dyn |> Dynparam.add Pool_common.Repo.Version.t version
+    , {sql|
+          firstname_version = $3
+        |sql} )
+    |> update_user_table
+  | Lastname (version, value) ->
+    let%lwt (_ : Sihl_user.t) =
+      update_sihl_user
+        pool
+        ~lastname:(value |> Pool_user.Lastname.value)
+        contact
+    in
+    ( dyn |> Dynparam.add Pool_common.Repo.Version.t version
+    , {sql|
+            lastname_version = $3
+          |sql} )
+    |> update_user_table
+  | Paused (version, value) ->
+    ( dyn
+      |> Dynparam.add Caqti_type.bool (value |> Pool_user.Paused.value)
+      |> Dynparam.add Pool_common.Repo.Version.t version
+    , {sql|
+              paused = $3,
+              paused_version = $4
+            |sql}
+    )
+    |> update_user_table
+  | Language (version, value) ->
+    ( dyn
+      |> Dynparam.add Caqti_type.(option Pool_common.Repo.Language.t) value
+      |> Dynparam.add Pool_common.Repo.Version.t version
+    , {sql|
+                language = $3,
+                language_version = $4
+              |sql}
```

Comment:
Manual formatting, I'd prefer all closing `|sql}` to have the same indentation as the opening ones

Reply:
👍